### PR TITLE
FIX: upgrade to v5 of gradle spring boot

### PR DESCRIPTION
**Pull request by markreesmoj**
Trying to upgrade spring boot plugin - not the easiest

Remember to fix logback config removing logAppender

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/config/TelemetryConfig.kt [hunk 1]
```diff
@@ -27,22 +20,7 @@ class TelemetryConfig {
 
   private class AppInsightKeyAbsentCondition : Condition {
     override fun matches(@NonNull context: ConditionContext, @NonNull metadata: AnnotatedTypeMetadata): Boolean {
-      return StringUtils.isEmpty(context.environment.getProperty("application.insights.ikey"))
+      return context.environment.getProperty("application.insights.ikey").isNullOrEmpty()
     }
   }
-
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/config/TelemetryConfig.kt:33]: Not sure what these were for

File: src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/config/web/ObserveFields.kt [hunk 2]
```diff
@@ -1,8 +1,6 @@
 package uk.gov.justice.digital.hmpps.courthearingeventreceiver.config.web
 
 import org.springframework.boot.context.properties.ConfigurationProperties
-import org.springframework.boot.context.properties.ConstructorBinding
 
-@ConstructorBinding
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/config/web/ObserveFields.kt:6]: Not sure what this did

**Review by Obsiye — DISMISSED** [review]

File: src/main/resources/logback-spring.xml [hunk 3]
```diff
@@ -3,59 +3,34 @@
...
-        <appender-ref ref="logAppender"/>
         <appender-ref ref="consoleAppender"/>
     </logger>
-
     <logger name="org.springframework" additivity="false" level="INFO">
-        <appender-ref ref="logAppender"/>
+        <appender-ref ref="consoleAppender"/>
     </logger>
-
     <logger name="com.microsoft.applicationinsights" additivity="false" level="INFO">
```
Comment (Obsiye) [inline src/main/resources/logback-spring.xml:24]: Should we have the app insights connection string in the secrets [more info](https://learn.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-upgrade-from-2x#step-3-configure-your-application-insights-connection-string). I don't know if the instrumentation key is enough 

I think logback is autoinstrumented for application insights so the section `<logger name="com.microsoft.applicationinsights"`  can be removed

**Review by Obsiye — COMMENTED** [review]
looks good to me. I left a comment about application insights but i'm ready to approve once addressed

**Review by Obsiye — APPROVED** [review]

**Pull request merged**